synapse/handlers/sync.py

```diff
@@ -565,10 +567,21 @@ def generate_sync_result(self, sync_config, since_token=None, full_state=False):
         # Always use the `now_token` in `SyncResultBuilder`
         now_token = yield self.event_sources.get_current_token()
 
+        user_id = sync_config.user.to_string()
+        app_service = self.store.get_app_service_by_user_id(user_id)
+        if app_service:
+            rooms = yield self.store.get_app_service_rooms(app_service)
```

Comment:
do we not need to worry about the same thing here?

Reply:
Yeeeeeeeeeeeeeeeeeees, though I really want to burn this code path with fire since its so inefficient and nothing should really be using it. Looking at the code again it actually may not be as bad to fix as I initially thought

Reply:
(I'll do that)

Reply:
Hmm, though I think this code paths is actually broken, as it tries to instantiate `RoomsForUser` with too few arguments.

Reply:
i.e., I'm wondering if we can just stub it out with a `NotImplementedError` or something